config-provisioning/src/main/java/com/yahoo/config/provision/SystemName.java

```diff
@@ -66,12 +71,23 @@ public String value() {
     // TODO Remove and use isPublicLike() instead
     public boolean isPublic() { return isPublicLike(); }
 
-    public boolean isMainLike() { return this == main || this == cd; }
+    // TODO Remove and use isYahoo() instead
+    public boolean isMainLike() { return isYahoo(); }
+
+    public boolean isYahoo() { return this == main || this == cd; }
```

Comment:
I don't know - main is also the default system so it would be wrong to call it Yahoo for those cases.  And without also changing HostedSystem and most references to 'main' this is probably more confusing than not.

And why migrate 2 methods to *Like, but then introduce 2 methods not named *Like?

Reply:
> I don't know - main is also the default system so it would be wrong to call it Yahoo for those cases. And without also changing HostedSystem and most references to 'main' this is probably more confusing than not.

I can take a stab at adding a `default` system and changing the default.

> And why migrate 2 methods to *Like, but then introduce 2 methods not named *Like?

The idea was to only use the `Like()` suffix where the prefix collides with a system name - `isCd()` is ambiguous but `isProd()` is not.

Reply:
Another possibility is to call `main` "Yahoo's main system" and `CD` "Yahoo's CD system".  In types and identifiers one could write isYahooMain etc, which makes it easier to understand `main` wherever that is seen, but still get the point across that it is a Yahoo system.

Reply:
It should in theory be safe to rename the enum variants, e.g. `main` => `yahooMain`, but I suspect we do not use `.value()` / `.from(String)` in all our (de)serialization code paths.